src/axp/java/com/lambdaherding/edi/axp/ch03/StreamExample.java

```diff
@@ -0,0 +1,52 @@
+package com.lambdaherding.edi.axp.ch03;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StreamExample {
+	/*
+	 * Exercise 1.1
+	 */
+	public static int addUp( Stream<Integer> numbers ) {
+		return numbers.collect( Collectors.summingInt( i -> i ) );
+	}
+
+	/*
+	 * Exercise 6 (this works by magic.)
+	 */
+	public static long countLower( String str ) {
+		return str.chars().filter( i -> Math.abs( i - 109.5 ) < 13 ).count();
+	}
+
+	/*
+	 * Exercise 7
+	 */
+	public static Optional<String> mostLowercase( List<String> strlist ) {
+		return strlist.stream().max( Comparator.comparing( s -> countLower( s ) ) );
```

Comment:
We've not got to method references in the chapter yet, but this could be written:
```java
Comparator.comparing( StreamExample::countLower )
```